src/cbuild/hooks/post_install/007_collect_debug_sources.py

```diff
@@ -0,0 +1,67 @@
+from cbuild.core import chroot
+
+import pathlib
+import shutil
+
+
+def _scan_file(pkg, v, processed):
+    # silently ignore invalid files
+    for source_file in (
+        chroot.enter(
+            "list-dwarf-srcfiles",
+            f"/{v.relative_to(pkg.rparent.bldroot_path)}",
+            capture_output=True,
+            ro_root=True,
+            ro_build=True,
+            unshare_all=True,
+        )
+        .stdout.strip()
+        .splitlines()
+    ):
+        source_file = pathlib.Path(source_file.strip().decode())
+
+        # don't have access to these!
+        if source_file.name in ["<stdin>"]:
+            continue
+
+        # /usr sourcefiles like /usr/include headers are already provided by -devel packages
+        if source_file.is_relative_to("/usr"):
+            continue
+        if source_file in processed:
+            # already seen
+            continue
+        processed.add(source_file)
+
+        pkg_prefix = pathlib.Path(
+            f"/usr/src/debug/{pkg.rparent.pkgname}-{pkg.rparent.pkgver}-r{pkg.rparent.pkgrel}"
+        )
+        if not source_file.is_absolute():
+            src = pkg.rparent.srcdir / source_file
+
+            if not src.exists():
+                pkg.log_warn(f"missing debug source file: {source_file}")
+            else:
+                dst = (
+                    pkg.rparent.destdir_base
+                    / f"{pkg.rparent.pkgname}-dbg-src-{pkg.rparent.pkgver}"
+                    / source_file
+                )
+                dst.parent.mkdir(parents=True, exist_ok=True)
+                shutil.copy(src, dst)
+        else:
+            pkg.log_warn(f"unknown debug source file: {source_file}")
```

Comment:
as a guarantee, every single build is going to have (picked from a random exe):

```
./build/../crt/Scrt1.c
./build/../crt/crt1.c
./build/runtimes/builtins-bins/lib/builtins/./compiler-rt/lib/builtins/crtbegin.c
```

this is because these are referenced by the linked `crt` objects (there is another path for static-linked executables, and maybe another for dsos(?)). they will always be missing except in the build that makes the actual crt (i.e. clang-rt-dbg-src(?) for compiler-rt and musl-dbg-src for libc crt). having a warning printed in every single build is not super great so perhaps there should be some ignored filepaths, but only for ignoring this warning (since ignoring the path whole will prevent the -dbg-src from having it)

Reply:
probably like

```
elif source_file.name in ["Scrt1.c", "crt1.c", .....]:
    # doesn't exist, but we don't warn these because they come from the toolchain
```